### PR TITLE
Apply Gradle Enterprise Conventions plugin

### PR DESCRIPTION
**Pull request by blindpirate**
This PR extract command build cache/build scan configuration to an isolated plugin, which is aimed at providing GE supports for all gradle public projects via public GE instance.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test CA

**Comment by blindpirate** [comment]
@bot-teamcity test QF

**Comment by blindpirate** [comment]
@bot-teamcity test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@bot-gradle test QF

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt [hunk 1]
```diff
@@ -35,7 +35,6 @@ val m2CleanScriptUnixLike = """
         tree ${'$'}REPO
         rm -rf ${'$'}REPO
         echo "${'$'}REPO was polluted during the build"
-        exit 1
```
Comment (blindpirate) [inline .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt:38]: This is just to make TeamCity build happy, it won't happen.

File: .teamcity/common/extensions.kt [hunk 2]
```diff
@@ -91,8 +91,34 @@ fun BuildType.applyDefaultSettings(os: Os = Os.linux, timeout: Int = 30, vcsRoot
             param("env.LC_ALL", "en_US.UTF-8")
         }
     }
+
+    publishConventionsPlugin(os)
+
+}
+
+
+private fun BuildType.publishConventionsPlugin(os: Os) {
```
Comment (big-guy) [inline .teamcity/common/extensions.kt:100]: Why not just use an included build?

Comment (jjohannes) [reply .teamcity/common/extensions.kt:100]: We can't do that atm because included builds break the configuration cache.

I am working on restructuring/cleanup of our buildSrc. So that we can easily switch to included builds once they are supported. Maybe this should then also just become a buildSrc plugin for now and later we move all of them to included builds.
Or does buildSrc not work for setting plugins?

File: gradle-enterprise-conventions-plugin/gradlew [hunk 3]
```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env sh
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/gradlew:1]: :o: I wonder if we should just use the wrapper from the main project here to avoid multiple places that require updates. 

This can be done by replacing this wrapper script here with:

```
  
#!/usr/bin/env sh

exec "$(dirname "$0")/../gradlew" "$@"
```

File: gradle-enterprise-conventions-plugin/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    jcenter()
+    gradlePluginPortal()
+}
+
+dependencies {
+    val gradleEnterprisePluginVersion = "3.3.4"
+    compileOnly("com.gradle:gradle-enterprise-gradle-plugin:${gradleEnterprisePluginVersion}")
+    implementation(gradleApi())
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/build.gradle.kts:21]: :x: We should use the short notation for Kotlin dependencies `implementation(kotlin(stdlib-jdk8))`.

File: gradle-enterprise-conventions-plugin/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+repositories {
+    jcenter()
+    gradlePluginPortal()
+}
+
+dependencies {
+    val gradleEnterprisePluginVersion = "3.3.4"
+    compileOnly("com.gradle:gradle-enterprise-gradle-plugin:${gradleEnterprisePluginVersion}")
+    implementation(gradleApi())
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/build.gradle.kts:20]: Is the 'kotlin-bom' needed here? I thought the versions are added by the Kotlin plugin anyway?

File: gradle-enterprise-conventions-plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+You can install the plugin to local maven repository via:
+
+```
+./gradlew publishPluginMavenPublicationToMavenLocal
+```
+
+Then use the plugin under development via:
+
+```
+buildscript {
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/README.md:78]: This should only use then also only use the plugins block (see other comment)

File: gradle-enterprise-conventions-plugin/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+To enable build scan publishing, you need to correctly authenticate as documeneted [here](https://docs.gradle.com/enterprise/gradle-plugin/#authenticating_with_gradle_enterprise).
+
+## Development
+
+Feel free to fork this repository, customize the plugin, and make contribution!
+
+You can install the plugin to local maven repository via:
+
+```
+./gradlew publishPluginMavenPublicationToMavenLocal
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/README.md:72]: The name of this task will be different based on the name of the local repository (see other comment).

File: gradle-enterprise-conventions-plugin/README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
+# Gradle Enterprise Helper Plugin
+
+Inspired by https://github.com/spring-gradle-plugins/gradle-enterprise-conventions-plugin , this plugin configure public [Gradle projects](https://github.com/gradle)
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/README.md:3]:
```suggestion
Inspired by https://github.com/spring-gradle-plugins/gradle-enterprise-conventions-plugin, this plugin configures public [Gradle projects](https://github.com/gradle)
```

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+val isCiServer = System.getenv().containsKey("CI")
+val gradleEnterpriseServerUrl = System.getProperty("gradle.enterprise.url") ?: "https://ge.gradle.org"
+val remoteCacheUrl = System.getProperty("gradle.cache.remote.url")?.let { URI(it) } ?: determineCacheNode()
+val remotePush = System.getProperty("gradle.cache.remote.push") != "false"
+val remoteCacheUsername = System.getProperty("gradle.cache.remote.username", "")
+val remoteCachePassword = System.getProperty("gradle.cache.remote.password", "")
+val disableLocalCache = System.getProperty("disableLocalCache")?.toBoolean() ?: false
+
+fun determineCacheNode(): URI {
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt:25]:
```suggestion
fun determineCacheNode() =
```

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        TravisCICustomValueProvider,
+        JenkinsCiCustomValueProvider,
+        GitHubActionsCustomValueProvider,
+        TeamCityCustomValueProvider,
+        LocalBuildCustomValueProvider,
+        GitInformationCustomValueProvider
+    )
+
+    override fun apply(settings: Settings) {
+        settings.plugins.withType(GradleEnterprisePlugin::class.java) {
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt:46]:
```suggestion
    override fun apply(settings: Settings) = settings.plugins.withType<GradleEnterprisePlugin>() {
```

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+                settings.buildCache(BuildCacheConfigureAction())
+            }
+            if (!settings.gradle.startParameter.isNoBuildScan) {
+                BuildScanConfigureAction(buildScanCustomValueProviders).execute(settings.extensions.getByType(GradleEnterpriseExtension::class.java).buildScan)
+            }
+        }
+    }
+}
+
+class BuildCacheConfigureAction : Action<BuildCacheConfiguration> {
```
Comment (jjohannes) [inline gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt:57]: :x: No need to implement an Action interface in Kotlin. Instead I would use an extension method here. That would be more idiomatic.

`private fun BuildCacheConfiguration.configureBuildCache() { ... }`

(same for BuildScanConfigureAction below)

File: settings.gradle.kts [hunk 12]
```diff
@@ -23,11 +23,18 @@ pluginManagement {
...
+        classpath("org.gradle.enterprise:gradle-enterprise-conventions-plugin:0.1-SNAPSHOT")
+    }
+}
+
 plugins {
     id("com.gradle.enterprise").version("3.3.4")
 }
 
-apply(from = "gradle/build-cache-configuration.settings.gradle.kts")
+apply(plugin = "org.gradle.enterprise.gradle-enterprise-conventions-plugin")
```
Comment (jjohannes) [inline settings.gradle.kts:37]: :x: We should use `plugins {}` block (and `pluginManagement {}`) instead of `buildscript` and `apply(plugin = ...)`.

:x: We should not use `mavenLocal()`. It is a global state on the machine used for different things, which can lead to a ton of problems. We really only support `mavenLocal()` in Gradle for interoperability with Maven.

Instead, define a local repository that we use. Like `maven { name = "pluginsLocal"; url = "local-plugins-repo") }` and add the repo folder to .gitignore

**Comment by blindpirate** [comment]
@bot-gradle test QFL

File: gradle-enterprise-conventions-plugin/gradlew (hunk 3 above)
Comment (blindpirate) [reply gradle-enterprise-conventions-plugin/gradlew]: This is moved to a separate repository: https://github.com/gradle/gradle-enterprise-conventions-plugin

File: gradle-enterprise-conventions-plugin/build.gradle.kts (hunk 4 above)
Comment (blindpirate) [reply gradle-enterprise-conventions-plugin/build.gradle.kts]: It's no longer Kotlin.

File: gradle-enterprise-conventions-plugin/README.md (hunk 7 above)
Comment (blindpirate) [reply gradle-enterprise-conventions-plugin/README.md]: This will be resolved in the new repo.

**Comment by blindpirate** [comment]
@bot-gradle test QFL

File: .gitignore [hunk 13]
```diff
@@ -1,5 +1,6 @@
 # Gradle
 # ------
+gradle-enterprise-conventions-plugin/build
```
Comment (wolfs) [inline .gitignore:3]: Why add this to `.gitignore`?

File: .teamcity/common/BuildCache.kt [hunk 14]
```diff
@@ -20,18 +20,19 @@ interface BuildCache {
     fun gradleParameters(os: Os): List<String>
 }
 
-data class RemoteBuildCache(val url: String, val username: String = "%gradle.cache.remote.username%", val password: String = "%gradle.cache.remote.password%") : BuildCache {
+data class RemoteBuildCache(val url: String,
+                            val username: String = "%gradle.cache.remote.public.username%",
```
Comment (wolfs) [inline .teamcity/common/BuildCache.kt:24]: Does that mean that after this PR, all developers need to switch to use the public build cache instance?

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 15]
```diff
@@ -65,11 +64,9 @@ open class BuildScanPlugin : Plugin<Project> {
     override fun apply(project: Project): Unit = project.run {
         buildScan = the()
 
-        extractCiOrLocalData()
-        extractVcsData()
+        extractCIData()
```
Comment (wolfs) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:67]:
```suggestion
        extractCiData()
```

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 16]
```diff
@@ -241,32 +208,6 @@ open class BuildScanPlugin : Plugin<Project> {
...
-                execAndGetStdout("git", "rev-parse", "--abbrev-ref", "HEAD").takeIf { it.isNotEmpty() && it != "HEAD" }?.let { branchName ->
-                    tag(branchName)
-                    value("Git Branch Name", branchName)
-                }
-            }
-        }
-    }
-
     private
     fun Project.extractBuildCacheData() {
```
Comment (wolfs) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:212]: Can we move that to the plugin as well? `extractWatchFsData` would also be great for the plugin, though this can be done later as well.

File: gradle/init-scripts/build-scan.init.gradle.kts [hunk 17]
```diff
@@ -27,10 +27,5 @@ fun configureExtension(extension: Any) {
         } else {
             "publishAlways"()
         }
-        setProperty("server", "https://e.grdev.net")
-
-        if (!System.getProperty("slow.internet.connection", "false").toBoolean()) {
-            setProperty("captureTaskInputFiles", true)
```
Comment (wolfs) [inline gradle/init-scripts/build-scan.init.gradle.kts:33]: Why remove this? I don't see the plugin activating `captureTaskInputFiles`.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 18]
```diff
@@ -188,48 +185,18 @@ open class BuildScanPlugin : Plugin<Project> {
     fun isEc2Agent() = java.net.InetAddress.getLocalHost().hostName.startsWith("ip-")
 
     private
-    fun Project.extractCiOrLocalData() {
+    fun Project.extractCIData() {
         if (isCiServer) {
             buildScan {
-                tag("CI")
+                setCompileAllScanSearch(execAndGetStdout("git", "rev-parse", "--verify", "HEAD"))
```
Comment (wolfs) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:191]: Are we running that twice now? Once for the plugin and once here? Wouldn't it be nicer to expose the commit id from the plugin, so it can be used when building on top of it?

File: .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt [hunk 19]
```diff
@@ -23,5 +23,8 @@ open class BaseGradleBuildType(model: CIBuildModel, val stage: Stage? = null, us
                 artifacts(days = 7)
             }
         }
+        params {
+            param("env.GRADLE_ENTERPRISE_ACCESS_KEY", "%ge.gradle.org.access.key%")
```
Comment (wolfs) [inline .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt:27]: Do we need to pass that key now for every build? Why don't we set it on the parent project?

File: .teamcityTest/Gradle_Check_Tests/ApplyDefaultConfigurationTest.kt [hunk 20]
```diff
@@ -14,6 +14,7 @@ import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.GradleBuildStep
 import model.CIBuildModel
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
```
Comment (wolfs) [inline .teamcityTest/Gradle_Check_Tests/ApplyDefaultConfigurationTest.kt:17]: What happened here? Is this an unused import?

File: .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt (hunk 19 above)
Comment (etiennestuder) [reply .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt:27]: We should not set such an access key at all. But instead use the API that allows to publish automatically when authenticated - just like Spring does it. This will also make it very concenient for building locally, either as a Gradle developer with an access (scans will be published) or an external developer (no scans will be published). @facewindu can give more insights if needed.

Comment (etiennestuder) [reply .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt:27]: Actually, I just realized this is on the TC config. There, it might make sense to set it on the top-most parent project. The rest I wrote about automatic publishing only when authenticated remains.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 21]
```diff
@@ -65,11 +64,9 @@ open class BuildScanPlugin : Plugin<Project> {
     override fun apply(project: Project): Unit = project.run {
         buildScan = the()
 
-        extractCiOrLocalData()
-        extractVcsData()
+        extractCIData()
 
         if (isCiServer) {
-            doNotUploadInBackground()
```
Comment (etiennestuder) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:72]: Why was this removed?

File: gradle-enterprise-conventions-plugin/README.md [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+    dependencies {
+        classpath("com.github.blindpirate:gradle-enterprise-helper-plugin:${thePluginVersion}")
+    }
+}
+
+plugins {
+    id("com.gradle.enterprise").version("3.3.4")
+}
+
+apply(plugin= "com.github.blindpirate.gradle-enterprise-helper")
```
Comment (etiennestuder) [inline gradle-enterprise-conventions-plugin/README.md]: I would not use 'helper' in the plugin name, but 'convention'.

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+    override fun execute(buildCache: BuildCacheConfiguration) {
+        buildCache.remote(HttpBuildCache::class.java) { remoteBuildCache ->
+            remoteBuildCache.url = remoteCacheUrl
+            remoteBuildCache.isPush = isCiServer && remotePush
```
Comment (etiennestuder) [inline gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt]: Why do you need the ability to to disable pushing on CI?

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+                remoteBuildCache.credentials {
+                    it.username = remoteCacheUsername
+                    it.password = remoteCachePassword
+                }
+            }
+        }
+
+        if (disableLocalCache) {
+            buildCache.local { localBuildCache ->
+                localBuildCache.isEnabled = false
```
Comment (etiennestuder) [inline gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt]: localBuildCache.isEnabled = !disableLocalCache

File: .gitignore (hunk 13 above)
Comment (blindpirate) [reply .gitignore]: It was a leftover. Removed.

File: .teamcity/common/BuildCache.kt (hunk 14 above)
Comment (blindpirate) [reply .teamcity/common/BuildCache.kt:24]: No, this is just for transition from `e.grdev.net` to `ge.gradle.org`. After this is merged, we'll keep this config for some time until all developer branches catch up. Then we switch to old config name.

File: gradle/init-scripts/build-scan.init.gradle.kts (hunk 17 above)
Comment (blindpirate) [reply gradle/init-scripts/build-scan.init.gradle.kts:33]: It's here: https://github.com/gradle/gradle-enterprise-conventions-plugin/blob/9e0fc6545148014273ef2fd3e1c94beacfb7fb54/src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java#L64

I don't know if `slow.internet.connection` is still useful.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 18 above)
Comment (blindpirate) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:191]: Yes, the command is run twice. We can probably store the information in the plugin.

File: .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt (hunk 19 above)
Comment (blindpirate) [reply .teamcity/Gradle_Check/configurations/BaseGradleBuildType.kt]: I put it in `RootProject`.

File: .teamcityTest/Gradle_Check_Tests/ApplyDefaultConfigurationTest.kt (hunk 20 above)
Comment (blindpirate) [reply .teamcityTest/Gradle_Check_Tests/ApplyDefaultConfigurationTest.kt]: It was a leftover. I removed locally and forgot to push.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 21 above)
Comment (blindpirate) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:72]: The logic is moved to plugin code: https://github.com/gradle/gradle-enterprise-conventions-plugin/blob/9e0fc6545148014273ef2fd3e1c94beacfb7fb54/src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java#L67

File: gradle-enterprise-conventions-plugin/README.md (hunk 22 above)
Comment (blindpirate) [reply gradle-enterprise-conventions-plugin/README.md]: It's removed.

File: gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt (hunk 24 above)
Comment (blindpirate) [reply gradle-enterprise-conventions-plugin/src/main/kotlin/org/gradle/enterprise/GradleEnterpriseConventionsPlugin.kt]: Already done here: https://github.com/gradle/gradle-enterprise-conventions-plugin/blob/9e0fc6545148014273ef2fd3e1c94beacfb7fb54/src/main/java/com/gradle/enterprise/conventions/GradleEnterpriseConventionsPlugin.java#L106

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by lptr** [comment]
Let's call this something other than "GE plugin" as that name is already taken. "GE configuration plugin" maybe?

File: gradle/init-scripts/build-scan.init.gradle.kts (hunk 17 above)
Comment (wolfs) [reply gradle/init-scripts/build-scan.init.gradle.kts:33]: I don't know how bad this affects folks when using background upload. Maybe @melix has some input.

File: .teamcity/common/BuildCache.kt (hunk 14 above)
Comment (wolfs) [reply .teamcity/common/BuildCache.kt:24]: I don't understand. After this is merged, CI will push to the new cache location. So if developers want to get some cache hits, they will need to use the new cache URL, right?

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 18 above)
Comment (wolfs) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt:191]: This should also happen in the background, if not it will block execution of the project.

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@lptr `ge-plugin` is just the branch name. The plugin itself is called https://github.com/gradle/gradle-enterprise-conventions-plugin

File: .teamcity/common/BuildCache.kt (hunk 14 above)
Comment (blindpirate) [reply .teamcity/common/BuildCache.kt]: I see your point. I already changed global cache configuration to `ge.gradle.org` and sent an email ton engineering maillist.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 16 above)
Comment (blindpirate) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Done with `0.2` plugin version.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 18 above)
Comment (blindpirate) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Done.

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
@bot-gradle test this

File: gradle/init-scripts/build-scan.init.gradle.kts (hunk 17 above)
Comment (melix) [reply gradle/init-scripts/build-scan.init.gradle.kts:33]: Slow internet connection is for downloads from the cache, when it's not worth downloading. For upload it's not a problem in my experience thanks to the async upload.
